sw/device/lib/dif/register.c

```diff
@@ -0,0 +1,16 @@
+// Copyright lowRISC contributors.
```

Comment:
@mcy I have split out this change, but before I create a new PR, could you please expand a little on why register.c is necessary?  What benefits does it bring as opposed to just having register.h?

Reply:
These declarations give the symbols real addresses, otherwise they're purely inline, AIUI.  I'd include them for the sake of completeness.

Reply:
They're required so that you (and the compiler) can write `&reg32_write` and so on. I don't believe you can get away with not including them; in theory, leaving them off is UB, and, in practice, results in linker errors based on how inline-ey the compiler feels that morning.

It's less a completeness thing and more a "ISO C requires this" thing. @lenary can call me out if I'm wrong though.

Reply:
Thanks @mcy and @gkelly

Reply:
> They're required so that you (and the compiler) can write `&reg32_write` and so on. I don't believe you can get away with not including them; in theory, leaving them off is UB, and, in practice, results in linker errors based on how inline-ey the compiler feels that morning.
> 
> It's less a completeness thing and more a "ISO C requires this" thing. @lenary can call me out if I'm wrong though.

Thanks for the clarification!